### PR TITLE
add ipynb - text transformers

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
adding https://github.com/PyTorchLightning/pytorch-lightning/blob/master/notebooks/04-transformers-text-classification.ipynb

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: starters/transformers-text-classification/text-transformers.py [hunk 1]
```diff
@@ -0,0 +1,612 @@
...
+    "MNLI": URL_DATA + "%2FMNLI.zip?alt=media&token=50329ea1-e339-40e2-809c-10c40afff3ce",
+    "SNLI": URL_DATA + "%2FSNLI.zip?alt=media&token=4afcfbb2-ff0c-4b2d-a09a-dbf07926f4df",
+    "QNLI": URL_DATA + "%2FQNLIv2.zip?alt=media&token=6fdcf570-0fc5-4631-8456-9505272d1601",
+    "RTE": URL_DATA + "%2FRTE.zip?alt=media&token=5efa7e85-a0bb-4f19-8ea2-9e1840f077fb",
+    "WNLI": URL_DATA + "%2FWNLI.zip?alt=media&token=068ad0a0-ded7-4bd7-99a5-5e00222e0faf",
+    "diagnostic": [
+        "https://storage.googleapis.com/mtl-sentence-representations.appspot.com/tsvsWithoutLabels%2FAX.tsv?GoogleAccessId=firebase-adminsdk-0khhl@mtl-sentence-representations.iam.gserviceaccount.com&Expires=2498860800&Signature=DuQ2CSPt2Yfre0C%2BiISrVYrIFaZH1Lc7hBVZDD4ZyR7fZYOMNOUGpi8QxBmTNOrNPjR3z1cggo7WXFfrgECP6FBJSsURv8Ybrue8Ypt%2FTPxbuJ0Xc2FhDi%2BarnecCBFO77RSbfuz%2Bs95hRrYhTnByqu3U%2FYZPaj3tZt5QdfpH2IUROY8LiBXoXS46LE%2FgOQc%2FKN%2BA9SoscRDYsnxHfG0IjXGwHN%2Bf88q6hOmAxeNPx6moDulUF6XMUAaXCSFU%2BnRO2RDL9CapWxj%2BDl7syNyHhB7987hZ80B%2FwFkQ3MEs8auvt5XW1%2Bd4aCU7ytgM69r8JDCwibfhZxpaa4gd50QXQ%3D%3D",  # noqa
+        "https://www.dropbox.com/s/ju7d95ifb072q9f/diagnostic-full.tsv?dl=1",
+    ],
+}
```
Comment (Borda) [inline starters/transformers-text-classification/text-transformers.py:89]: @SeanNaren any idea about these links, how I am getting forbidden access :confused:

File: lightning_examples/transformers-text-classification/.meta.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+title: Finetune Transformers Models with PyTorch Lightning
+author: PL team
+created: 2021-01-31
+updated: 2021-02-23
```
Comment (Borda) [inline lightning_examples/transformers-text-classification/.meta.yml:4]:
```suggestion
updated: 2021-06-21
```

File: lightning_examples/transformers-text-classification/.meta.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+license: CC
```
Comment (Borda) [inline lightning_examples/transformers-text-classification/.meta.yml:5]:
```suggestion
license: CC BY-SA
```

File: lightning_examples/transformers-text-classification/text-transformers.py [hunk 4]
```diff
@@ -0,0 +1,336 @@
...
+
+# %% [markdown] colab_type="text" id="fqlsVTj7McZ3"
+# # Training BERT on the GLUE Dataset
+
+# %% [markdown] id="7uQVI-xv9Ddj"
+# ---
+# ## Install & Imports
+
+# %% colab={} colab_type="code" id="OIhHrRL-MnKK"
+# !pip install pytorch-lightning datasets transformers
```
Comment (Borda) [inline lightning_examples/transformers-text-classification/text-transformers.py:25]:
```suggestion
```

File: lightning_examples/transformers-text-classification/text-transformers.py [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+        self.task_name = task_name
+        self.max_seq_length = max_seq_length
+        self.train_batch_size = train_batch_size
+        self.eval_batch_size = eval_batch_size
+
+        self.text_fields = self.task_text_field_map[task_name]
+        self.num_labels = self.glue_task_num_labels[task_name]
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path, use_fast=True)
+
+    def setup(self, stage):
```
Comment (Borda) [inline lightning_examples/transformers-text-classification/text-transformers.py:95]:
```suggestion
    def setup(self, stage: str):
```

**Review by SeanNaren — APPROVED** [review]

File: lightning_examples/text-transformers/.meta.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+title: Finetune Transformers Models with PyTorch Lightning
+author: PL team
+created: 2021-01-31
+updated: 2021-06-21
+license: CC BY-SA
+description: |
+  This notebook will use HuggingFace's `datasets` library to get data, which will be wrapped in a `LightningDataModule`.
+  Then, we write a class to perform text classification on any dataset from the[ GLUE Benchmark](https://gluebenchmark.com/).
```
Comment (ethanwharris) [inline lightning_examples/text-transformers/.meta.yml:8]:
```suggestion
  Then, we write a class to perform text classification on any dataset from the [GLUE Benchmark](https://gluebenchmark.com/).
```

File: lightning_examples/text-transformers/text-transformers.py [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+    task_name=dm.task_name
+)
+
+trainer = Trainer(max_epochs=3, gpus=min(1, torch.cuda.device_count()))
+trainer.fit(model, dm)
+
+# %% [markdown] colab_type="text" id="iZhbn0HzfdCu"
+# #### MNLI
+#
+#  - The MNLI dataset is huge, so we aren't going to bother trying to train it here.
```
Comment (ethanwharris) [inline lightning_examples/text-transformers/text-transformers.py:309]:
```suggestion
#  - The MNLI dataset is huge, so we aren't going to bother trying to train on it here.
```

**Review by ethanwharris — APPROVED** [review]
LGTM, small comments

**Pull request merged**